app/controllers/users/two_step_verifications_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Users::TwoStepVerificationsController < ApplicationController
+  def new
+    redirect_to two_step_verification_path if current_user.two_factor_auth_enabled?
```

Comment:
But it does not do any authorisation? It redirects to the `show` method in this class? (Which is used to show the status of the `two_step_verification` :-)